### PR TITLE
Don't use InitializeBuildTool for dotnet executable location.

### PR DESCRIPTION
**Pull request by 333fred**
In hindsight I should have actually physically tested https://github.com/dotnet/roslyn/pull/43682 intead of just reading the code.

### CODE REVIEW

File: scripts/vscode-build.ps1 [hunk 1]
```diff
@@ -15,9 +15,11 @@ if ($projectFileInfo) {
   $buildTool = InitializeBuildTool
   $frameworkArg = if ($framework -ne "") { " -p:TargetFramework=$framework" } else { "" }
   $buildArgs = "$($buildTool.Command) -v:m -m -p:UseRoslynAnalyzers=false -p:GenerateFullPaths=true$frameworkArg $($projectFileInfo.FullName)"
+  # InitializeBuildTool is hard coded for dotnet.exe, so use Ensure-DotnetSdk as it is xplat aware.
+  $buildPath = Ensure-DotnetSdk
```
Comment (RikkiGibson) [inline scripts/vscode-build.ps1:19]: Does the 'msbuild current project skip analyzers' task still work as expected?

Comment (333fred) [reply scripts/vscode-build.ps1:19]: I don't know, I can't test it.

Comment (RikkiGibson) [reply scripts/vscode-build.ps1]: Just to be clear, I pulled this down and manually tested on Windows. 👍

File: scripts/vscode-build.ps1 [hunk 2]
```diff
@@ -16,8 +16,16 @@ if ($projectFileInfo) {
   $frameworkArg = if ($framework -ne "") { " -p:TargetFramework=$framework" } else { "" }
   $buildArgs = "$($buildTool.Command) -v:m -m -p:UseRoslynAnalyzers=false -p:GenerateFullPaths=true$frameworkArg $($projectFileInfo.FullName)"
 
-  Write-Host "$($buildTool.Path) $buildArgs"
-  Exec-Console $buildTool.Path $buildArgs
+  $buildToolPath = $buildTool.Path
+
+  if ($msbuildEngine -ne "vs") {
+    # InitializeBuildTool is hardcoded to dotnet.exe. If we're not building using vs (meaning msbuild), use
```
Comment (jaredpar) [inline scripts/vscode-build.ps1:22]: > InitializeBuildTool is hardcoded to dotnet.exe

Why do you believe this is the case? The code for `InitializeBuildTool` will definitely use `msbuild` when the build engine is `"vs"` and it's pretty easy to demonstrate

```ps
P:\roslyn> $msbuildengine = "vs"
P:\roslyn> . .\eng\common\tools.ps1
P:\roslyn> $b = InitializeBuildTool
P:\roslyn> $b.Path
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\msbuild.exe
P:\roslyn> 
```

**Pull request closed without merging**